packages/dd-trace/src/span_sampler.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+const { globMatch } = require('../src/util')
+const {
+  USER_KEEP,
+  AUTO_KEEP
+} = require('../../../ext').priority
+const RateLimiter = require('./rate_limiter')
+
+class SpanSampler {
+  constructor ({ spanSamplingRules = [] }) {
+    this._rules = spanSamplingRules
+    this._limiters = {}
+  }
+
+  ingest (rootContext) {
```

Comment:
Isn't this the context of the actual span and not the root?

Reply:
Comes from the span, but it's used to extract the `context._trace.started` and then gets the context of each of those spans, so should probably be made clear it's something different. Not sure what a better name would be.

Reply:
That works with any span context from any span, so I guess just `spanContext` would be enough without giving it any special meaning, especially one that would be incorrect.

Reply:
Sure, but we should differentiate it somehow from the span contexts we are actually using for the analysis, not just the one for finding the list of spans.

Reply:
I've changed the naming to spanContext, though I feel the naming is a bit confusing given the use.